### ISSUE TITLE
Get rid of the "Customization" parent section

### ISSUE THREAD

**Issue by bentoi** [icerpc/icerpc-docs#233]
Looks good.

Not necessarily related to this PR but I think it might be a little confusing to users to have this customization section.

1. It gives the impression that these are these things you'd want to customize. Ie. tweak, modify. 

2. Now that everything is modular, these transports are not really any different to me than interceptors and middleware. It's just another module component you can add. Writing a middleware or interceptor is also custom. 

In summary, I think these pages just belong under connection or some other heading like "Protocol".

_Originally posted by @externl in https://github.com/icerpc/icerpc-docs/pull/226#pullrequestreview-1562904952_
